plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -56,19 +58,25 @@ def execute!(*_commands)
 
     private
 
-    # TODO only call once per cluster and filter the output
-    def pod_is_live?(release_doc)
+    def deployed_pods(release)
+      release.release_docs.map(&:deploy_group).flat_map do |deploy_group|
+        query = {
+          namespace: deploy_group.kubernetes_namespace,
+          label_selector: {
+            deploy_group_id: deploy_group.id,
+            project_id: @job.project_id,
+            release_id: release.id
+          }.map { |k, v| "#{k}=#{v}" }.join(',')
+        }
+        deploy_group.kubernetes_cluster.client.get_pods(query)
+      end
+    end
+
+    def pod_is_live?(pods, release_doc)
       group = release_doc.deploy_group
       role = release_doc.kubernetes_role.name
-      pod = group.kubernetes_cluster.client.get_pods(
-        namespace: group.kubernetes_namespace,
-        label_selector: {
-          deploy_group_id: group.id,
-          project_id: @job.project_id,
-          release_id: release_doc.kubernetes_release_id,
-          role: role
-        }.map { |k, v| "#{k}=#{v}" }.join(',')
-      ).first
+
+      pod = pods.detect { |pod| pod.metadata.labels.role == role && pod.metadata.labels.deploy_group_id.to_i == group.id }
```

Comment:
Maybe add a helper method to the `Pod` class to determine whether it's part of a ReleaseDoc?